### PR TITLE
Fix VfsPath Display and Debug impls

### PR DESCRIPTION
**Pull request by HKalbasi**
fix https://github.com/rust-lang/rust-analyzer/pull/15087#discussion_r1234544484
r? @matklad

### CODE REVIEW

File: crates/vfs/src/vfs_path.rs [hunk 1]
```diff
@@ -292,7 +292,7 @@ impl From<AbsPathBuf> for VfsPath {
 impl fmt::Display for VfsPath {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match &self.0 {
-            VfsPathRepr::PathBuf(it) => fmt::Debug::fmt(&it, f),
+            VfsPathRepr::PathBuf(it) => fmt::Display::fmt(&it, f),
```
Comment (jonas-schievink) [inline crates/vfs/src/vfs_path.rs:295]: Could also replace all of these (including the ones in the `Debug` impl below) with `it.fmt(f)`. Method resolution will prefer the right one.

Comment (matklad) [reply crates/vfs/src/vfs_path.rs:295]: I’d rather not. It’s is true that 

> Method resolution will prefer the right one

but this also is one of the least obvious rules in Rust.

no strong feelings here though!

Comment (matklad) [reply crates/vfs/src/vfs_path.rs:295]: Or maybe not! I think my main worry is that Debug&Disolay have a method with identical signature, which can be asily confused during method resolution. 

However, given that we never import the traits directly, and only use them through fmt:: prefix, using .fmt should be OK.

**Pull request merged**